sigsci_site_manager/util.py

```diff
@@ -0,0 +1,106 @@
+import json
+
+
+def filter_data(data, keys):
+    if isinstance(data, (list, tuple)):
+        ret = []
+        for item in data:
+            # Only save the data required by the create API
+            ret.append({k: item[k] for k in keys})
+    else:
+        ret = {k: data[k] for k in keys}
+    return ret
+
+
+def equal_conditions(a, b):
```

Comment:
Whew! It looks like this is just trying to compare two json strings to see if they're the same. You could do this by straight string comparison with a json dump: 

```
return json.dumps(a, sort_keys=True) == json.dumps(b, sort_keys=True)
```

That should even sort nested dictionaries.

Reply:
I tried a string comparison first and it doesn't work. `sort_keys` orders the keys within a dictionary but does nothing for the ordering of dictionaries within a list. With the SigSci API order doesn't matter within a list of conditions so you can have equivalent rules that dump to different string.

Reply:
Did you actually try out just using " a == b" for these dictionaries? I didn't give python enough credit, but it turns out it should be able to do these comparisons (even with nested dictionaries and lists) straight out of the box.

Reply:
```python
>>> a = {'type': 'group', 'conditions': [{'one', 1}, {'two': 2}]}
>>> b = {'type': 'group', 'conditions': [{'two', 2}, {'one', 1}]}
>>> a == b
False
```

Reply:
Those two dictionaries are not equal, so they should evaluate to false. In the second one, conditions contains 2 sets while in the first one in contains a set and a dictionary.

Reply:
The root problem is you can't sort a list of dicts because they're not comparable:
```python
>>> {'one': 1} < {'two': 2}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '<' not supported between instances of 'dict' and 'dict'
```

Reply:
> Those two dictionaries are not equal, so they should evaluate to false. In the second one, conditions contains 2 sets while in the first one it contains a set and a dictionary.

Yes, I see my typo. But the point remains that list ordering matters and these lists can't be sorted. We need to compare that these Signal Sciences rules are semantically the same even if their JSON representations are not identical. If you come up with a simpler way that works I'm all for it because this logic is very complex.